src/tqec/plaquette/rpng_description.py

```diff
@@ -0,0 +1,245 @@
+from tqec.circuit.schedule import ScheduledCircuit
+from tqec.circuit.qubit_map import QubitMap
+from tqec.plaquette.plaquette import Plaquette
+from tqec.plaquette.qubit import PlaquetteQubits
+from tqec.plaquette.qubit import SquarePlaquetteQubits
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional
+
+from stim import Circuit as stim_Circuit
+
+
+class BasisEnum(Enum):
+    X = 'x'
+    Y = 'y'
+    Z = 'z'
+
+
+class ExtendedBasisEnum(Enum):
+    X = BasisEnum.X.value
+    Y = BasisEnum.Y.value
+    Z = BasisEnum.Z.value
+    H = 'h'
+
+
+@dataclass
+class RPNG:
+    """Organize a single RPNG value
+
+    ----
+    Simplified RPNG format:
+
+    -z1-
+    rpng
+    
+    (r) data qubit reset basis or h or -  
+    (p) data basis for the controlled operation (x means CNOT controlled on the ancilla and targeting the data qubit, y means CY, z means CZ)  
+    (n) time step (positive integers, all distinct, typically in 1-5)  
+    (g) data qubit measure basis or h or -
+
+    Assumptions on the circuit:
+    - if not otherwise stated, a basis can be {x,y,z}
+    - the ancilla is always the control qubit for the CNOT, CY, or CZ gates
+    - time step of r same as ancilla reset (always 0)
+    - time step of g same as ancilla measurement (by default 6)
+
+    ----
+    Extended RPNG format:
+
+    The extra field (a) is used together with (p) to determine the 2Q gate involving the
+    ancilla and data qubits. For example:
+    - a=z, p=x --> CNOT controlled by the ancilla and targeting the data
+    - a=z, p=z --> CZ   controlled by the ancilla and targeting the data
+    - a=y, p=z --> CY   controlled by the data    and targeting the ancilla
+    By default a=z.
+    """
+    r: Optional[ExtendedBasisEnum]
+    p: Optional[BasisEnum]
+    n: Optional[int]
+    g: Optional[ExtendedBasisEnum]
+    a: BasisEnum = BasisEnum.Z
+
+    @classmethod
+    def from_string(cls, rpng_string: str) -> 'RPNG':
+        """Initialize the RPNG object from a 4-character or 5-character string
+        
+        4-character -> simplified format -> RPNG
+        5-character -> extended format -> RAPNG
+
+        Note that the ancilla basis(a) is the second character, just before (p).
+
+        Note that any character different from a BasisEnum / ExtendedBasisEnum
+        value would result in the corresponding field being None.
+        """
+        a_str = 'z'
+        if len(rpng_string) == 4:
+            r_str, p_str, n_str, g_str = tuple(rpng_string)
+        elif len(rpng_string) == 5:
+            r_str, a_str, p_str, n_str, g_str = tuple(rpng_string)
+        else:
+            raise ValueError('The rpng string must be exactly 4- or 5-character long.')
+        # Convert the characters into the enum attributes (or raise error).
+        r = ExtendedBasisEnum(r_str) if r_str in ExtendedBasisEnum._value2member_map_ else None
+        a = BasisEnum(a_str) if a_str in BasisEnum._value2member_map_ else None
```

Comment:
```suggestion
        a = BasisEnum(a_str) if a_str in BasisEnum else None
```
This should work.

Reply:
The suggestion does not work but alternatives may be:
`a_str in [item.value for item in BasisEnum]`
`BasisEnum(a_str) in BasisEnum`

If there is no specific issue, I will keep the original expression.